File: Лаба2/Задание 2 вариант 3/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	wh := make(map[int][]int)
	var keys []int
	col := map[byte]int{'R': 0, 'W': 1, 'B': 2}

	var a string
	fmt.Scanln(&a)

	for i := 0; i < len(a)/2; i++ {
		colc, _ := strconv.Atoi(string(a[i*2+1]))
		if _, ok := wh[colc]; ok {
			wh[colc][col[a[i*2]]] = 1
		} else {
			colors := make([]int, 3)
			colors[col[a[i*2]]] = 1
			wh[colc] = colors
			keys = append(keys, colc)
		}
	}

	count := 0
	for _, key := range keys {
		if b, ok := wh[key]; ok {
			if b[0]+b[1]+b[2] == 3 {
				count++
			}
		}
	}

	fmt.Println(count)
}

File: Лаба2/Задание 2 вариант 3/main.cpp
#include <vector>
#include <iostream>
#include <string>
#include <map>


using namespace std;

int main() {
	char bs;
	int colc;
	string a;
	map<int, vector<int>> wh;
	vector<int> keys;
	map<char, int> col = { {'R', 0}, {'W', 1}, {'B', 2} };
	cin >> a;
	for (int i = 0; i < a.size() / 2; i++) {
		if (!isalpha(a[i * 2])) {
			cout << "Error";
			return 1;
		}
		if (!isdigit(a[i * 2 + 1])) {
			cout << "Error";
			return 1;
		}
		if (a[i * 2] != 'R' && a[i * 2] != 'W' && a[i * 2] != 'B') {
			cout << "Error";
			return 1;
		}
		colc = a[i * 2 + 1] - '0';
		if (wh.find(colc) != wh.end()) {
			wh[colc][col[a[i * 2]]] = 1;
		}
		else {
			wh[colc] = { 0, 0, 0 };
			wh[colc][col[a[i * 2]]] = 1;
			keys.push_back(colc);
		}
	}
	int count = 0;
	vector<int> b;
	for (int i = 0; i < keys.size(); i++) {
		b = wh[keys[i]];
		if (b[0] + b[1] + b[2] == 3) {
			count++;
		}
	}
	cout << count << endl;
}
